probe/client/mongo/mongo.go

```diff
@@ -60,27 +64,50 @@ func New(opt conf.Options) Mongo {
 
 	tls, err := opt.TLS.Config()
 	if err != nil {
-		log.Errorf("[%s / %s / %s] - TLS Config error - %v", opt.ProbeKind, opt.ProbeName, opt.ProbeTag, err)
+		log.Errorf("[%s / %s / %s] - TLS Config Error - %v", opt.ProbeKind, opt.ProbeName, opt.ProbeTag, err)
+		return nil, fmt.Errorf("TLS Config Error - %v", err)
 	} else if tls != nil {
 		client.TLSConfig = tls
 		client.SetAuth(options.Credential{AuthMechanism: "MONGODB-X509"})
 	}
 
-	return Mongo{
+	mongo := &Mongo{
 		Options:   opt,
 		ConnStr:   conn,
 		ClientOpt: client,
 		Context:   context.Background(),
 	}
+
+	if err := mongo.checkData(); err != nil {
+		return nil, err
+	}
+
+	return mongo, nil
 }
 
 // Kind return the name of client
-func (r Mongo) Kind() string {
+func (r *Mongo) Kind() string {
 	return Kind
 }
 
+// checkData do the data checking
+func (r *Mongo) checkData() error {
+	if len(r.Data) > 0 {
+		for k, v := range r.Data {
+			if _, _, err := getDBCollection(k); err != nil {
+				return err
+			}
+			var bdoc interface{}
+			if err := bson.UnmarshalExtJSON([]byte(v), true, &bdoc); err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
propose to remove `if` to make code simpler.
```suggestion
	for k, v := range r.Data {
		if _, _, err := getDBCollection(k); err != nil {
			return err
		}
		var bdoc interface{}
		if err := bson.UnmarshalExtJSON([]byte(v), true, &bdoc); err != nil {
			return err
		}
	}
```